go/worker/commands.go

```diff
@@ -113,17 +123,33 @@ func upCmd(ctx *cli.Context) error {
 		if err != nil {
 			return err
 		}
+
+		uid := os.Getuid()
+		gid := os.Getgid()
+
 		// holds attributes that will be used when os.StartProcess.
-		// we use CLONE_NEWNS because ol creates many mount points.
-		// we don't want them to show up in /proc/self/mountinfo
-		// for systemd because systemd creates a service for each
-		// mount point, which is a major overhead.
+		// legacy used CLONE_NEWNS so mounts don't spam systemd.
+		// now, if --rootless, also create a user+UTS namespace and map uid/gid -> 0 (fake root).
 		attr := os.ProcAttr{
 			Files: []*os.File{nil, f, f},
-			Sys: &syscall.SysProcAttr{
-				Unshareflags: syscall.CLONE_NEWNS,
-			},
 		}
+		if rootless {
+			attr.Sys = &syscall.SysProcAttr{
```

Comment:
This code is the crux of what you're doing.  I think it needs some more commenting.  E.g., will it be obvious to people that "0" is the ID for root?  Also need to explain what a user namespace is, and why that lets us run rootless.